lambda/src/app.js

```diff
@@ -49,47 +55,37 @@ exports.lambdaHandler = async (event, context) => {
 
     const body = JSON.parse(event.body);
     let pet = body.pet;
-    if (event.body.image) {
-      // TODO invoke rekognition
-      // TODO set 'pet' to returned value
+    let imageBuffer = null;
+
+    if (body.image) {
+      imageBuffer = getImageBuffer(body.image);
+
+      const results = await rekognizeLabels(imageBuffer);
+      const supportedLabels = results.Labels.map(
+        (label) => label.Name
+      ).filter((name) => supportedPets.includes(name.toLowerCase()));
+
+      // Get matching supported pet label if available
+      if (supportedLabels.length > 0) {
+        pet = supportedLabels[0].toLowerCase();
+      } else {
+        // Get first label it has been identified as
+        pet = results.Labels[0].Name.toLowerCase();
+      }
     }
 
-    console.log(pet);
     if (supportedPets.indexOf(pet) > -1) {
       let id = uuidv4();
       let imagePath = null;
-      if (event.body.image) {
-        // TODO store in S3 as '<id>.<ext>'
-        // TODO set 'imagePath' to S3 URL
-      }
 
-      const item = {
-        id: { S: id },
-        pet: { S: pet },
-        imagePath: { S: imagePath },
-        userAgent: { S: event.headers["User-Agent"] },
-        desktopClient: {
-          BOOL: event.headers["CloudFront-Is-Desktop-Viewer"] === true,
-        },
-        mobileClient: {
-          BOOL: event.headers["CloudFront-Is-Mobile-Viewer"] === true,
-        },
-        smartTvClient: {
-          BOOL: event.headers["CloudFront-Is-SmartTV-Viewer"] === true,
-        },
-        tabletClient: {
-          BOOL: event.headers["CloudFront-Is-Tablet-Viewer"] === true,
-        },
-        country: { S: event.headers["CloudFront-Viewer-Country"] },
-        // TODO any other useful info that can be gleaned?
-      };
+      if (body.image) {
+        const key = id + "." + getFileType(body.image);
 
-      removeNullFields(item);
-      console.log(item);
+        await uploadImageToS3(bucketName, key, imageBuffer);
+        imagePath = getS3ObjectPath(bucketName, key, awsRegion);
```

Comment:
Can you use this path to retrieve the image?

Reply:
changed to put key (imageFileName) into dynamo